tests/Unit/Evolution/Systems/Cce/Test_Equations.cpp

```diff
@@ -0,0 +1,202 @@
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/Variables.hpp"  // IWYU pragma: keep
+#include "Evolution/Systems/Cce/Equations.hpp"
+#include "Evolution/Systems/Cce/Tags.hpp"  // IWYU pragma: keep
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+#include "tests/Unit/Pypp/CheckWithRandomValues.hpp"
+#include "tests/Unit/Pypp/SetupLocalPythonEnvironment.hpp"
+
+// IWYU pragma: no_forward_declare Cce::Tags::Beta
+// IWYU pragma: no_forward_declare Cce::Tags::H
+// IWYU pragma: no_forward_declare Cce::Tags::Q
+// IWYU pragma: no_forward_declare Cce::Tags::U
+// IWYU pragma: no_forward_declare Cce::Tags::W
+// IWYU pragma: no_forward_declare Cce::Tags::Integrand
+// IWYU pragma: no_forward_declare Cce::Tags::LinearFactor
+// IWYU pragma: no_forward_declare Cce::Tags::LinearFactorForConjugate
+// IWYU pragma: no_forward_declare Cce::Tags::PoleOfIntegrand
+// IWYU pragma: no_forward_declare Cce::Tags::RegularIntegrand
+// IWYU pragma: no_forward_declare Cce::TagsCategory::IndependentOfBondiIntegration
+// IWYU pragma: no_forward_declare Cce::TagsCategory::SwshDerivatives
+// IWYU pragma: no_forward_declare Cce::TagsCategory::TagsToSwshDifferentiate
+// IWYU pragma: no_forward_declare Cce::TagsCategory::Temporary
+// IWYU pragma: no_forward_declare Cce::ComputeBondiIntegrand
+// IWYU pragma: no_forward_declare Variables
+
+namespace Cce {
+
+namespace {
+
+template <typename Type, size_t N>
+using ForwardIgnoreIndex = Type;
+
+template <typename Tag, typename DataBoxType, typename Arg>
+int assign_to_tag(const gsl::not_null<DataBoxType*> box,
+                  const Arg& arg) noexcept {
+  db::mutate<Tag>(box,
+                  [&arg](const gsl::not_null<typename Tag::type*> box_value) {
+                    get(*box_value).data() = arg;
+                  });
+  // return value required for pack expansion
+  return 0;
+}
+
+template <typename DataBoxType, typename... ArgumentTags, typename... Args>
+void assign_to_tag_list(const gsl::not_null<DataBoxType*> box,
+                        tmpl::list<ArgumentTags...> /*meta*/,
+                        const Args&... args) noexcept {
+  expand_pack(assign_to_tag<ArgumentTags>(box, args)...);
+}
+
+// This is a wrapper struct such that the evaluate function can be called with
+// the full set of arguments that would ordinarily be packed into the
+// Variables. This is necessary for constructing a function that can be compared
+// to the pypp versions.
+template <typename Mutator, typename OutputTag, typename ArgumentTagList,
+          size_t NumberOfGridPoints, typename... Args>
+struct MutationFromArguments {
+  static ComplexDataVector evaluate(const Args&... args) noexcept {
+    // create a variables which will hold everything
+    using variables_tag =
+        ::Tags::Variables<tmpl::append<tmpl::list<OutputTag>, ArgumentTagList>>;
+
+    auto box = db::create<db::AddSimpleTags<variables_tag>>(
+        typename variables_tag::type{NumberOfGridPoints});
```

Comment:
I don't think you actually need to put everything into the `variables_tag`, you could also construct `db::AddSimpleTags` from a parameter pack.